.github/workflows/release.yml

```diff
@@ -0,0 +1,36 @@
+name: Release PECL Preparation
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+jobs:
+  package:
+    name: Prepare package
+    runs-on: macos-latest
+
+    steps:
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '8.0'
+
+    - name: Checkout Source
+      uses: actions/checkout@v3
+      with:
+          submodules: recursive
+          fetch-depth: 0
+
+    - name: Get release tag
+      uses: awslabs/aws-crt-builder/.github/actions/release-tag@main
```

Comment:
maybe we should move to something like the IoT SDKs use, where there's a release workflow they run that creates the tags

seems more explicit and less fragile than the current state of doing things where a release workflow runs in reaction to tags being created

Reply:
Related https://github.com/awslabs/aws-crt-php/blob/main/ext/crt.c#L275 from https://github.com/awslabs/aws-crt-php/issues/59

We should also update the version of the native extension. 

However, after I went through the IoT SDK actions, I think the issue from their approach is:
- Now they have two ways to cut a release: the github action and the old way of manually create a tag
- There is no enforcement to only use one approach and very likely, the old way will break stuff (Not running the script that initialized from github action)
- They already had multiple versions that are still cut manually and have different behavior for different tags, which is not good.

Yeah, it's more explicit, but easy to miss and break. I'd prefer to have only one place to trigger the actions.

Reply:
Maybe I'll remove this action to a separate PR to keep things more clear.